RadixWallet/Features/AppFeature/Overlay/Overlay+Reducer.swift

```diff
@@ -99,19 +103,19 @@ struct OverlayReducer: Sendable, FeatureReducer {
 	}
 
 	func reduceDismissedDestination(into state: inout State) -> Effect<Action> {
-		dismissAlert(state: &state, withAction: .dismissed)
+		if case let .alert(state) = state.itemsQueue.first {
+			overlayWindowClient.sendAlertAction(.dismissed, state.id)
```

Comment:
No, fullscreen can not be dismissed without sending a delegate action, and any delegate action will dismiss it.

Reply:
mm that goes for the Claim Wallet flow, but we probably will want to allow dismissal of other full screen flows presented this way.
We can make it so that it doesn't dismiss inside the view and just send `.delegate(.dismiss)` though

Reply:
The thing with fullscreen presentation is that you can't dismiss it by dragging or tapping outside, so you have to do it in code. But you are right, someone may think of calling the `dismiss()` dependency from the reducer, without actually sending any delegate actions, that wouldn't be good.

By the way, the way I set it up you have to send exactly one delegate action, and then we dismiss the view. Another way could have been to return a stream, and keep accepting delegate actions until the stream is completed, but that felt like overkill. For now you will have to do your logic either inside the fullscreen, or by telling the presenting reducer which delegate action to do. You can't for example continously update something in the presenting view, but that also is seldom needed when presenting a fullscreen view.

Reply:
I tested it and it work well. If someone uses TCA dismiss() then this method will get called, so I added handling here. Without that handling, the presenter would get stuck waiting for a delegate action forever.

Reply:
Noticed that we only _catch_ the first action, and I think that's fine. Still, although fullscreens cannot be dismissed by dragging or tapping outside, it could be that they have a `X` button on navigation bar or similar. In that case, there would need to be a `DelegateAction` that is `case canceled` and just be dismissed here, is that right?